timesheets/src/Code.js

```diff
@@ -0,0 +1,129 @@
+*/
+function onOpen() {
+  var ui = SpreadsheetApp.getUi();
+  ui.createMenu('Timesheets')
+      .addItem('Column Setup', 'columnSetup')
+      .addItem('Send Emails', 'sendEmails')
+      .addToUi();
+}
+
+/** Runs the "Column Setup" script. Adds "WEEKLY PAY" column with calculated
+ * values using array formulas. Adds an "APPROVAL" column at the end of
+ * the sheet, containing drop-down menus to either approve/disapprove employee
+ * timesheets.  Adds a "NOTIFIED STATUS" column indicating whether or not an
+ * employee has yet been e mailed.
+*/
+function columnSetup() {
+  // defines variables
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  var frozenRows = sheet.getFrozenRows();
+  var beginningRow = frozenRows + 1;
+
+  // adds new calculate pay column
+  sheet.insertColumnAfter(COLUMN_NUMBER.HOURLY_PAY);
+  sheet.getRange(1, COLUMN_NUMBER.CALC_PAY).setValue('WEEKLY PAY');
+
+  // calculates weekly pay using array formulas
+  sheet.getRange(beginningRow, COLUMN_NUMBER.CALC_PAY, lastRow - frozenRows, 1)
+      .setFormula('=ArrayFormula(SUM(D2:H2)) * I2:I');
+
+  // adds new approval column
+  sheet.insertColumnAfter(COLUMN_NUMBER.CALC_PAY);
+  sheet.getRange(1, COLUMN_NUMBER.APPROVAL).setValue('APPROVAL');
+
+  // make sure approval column is all drop-down menus
+  var approvalColumnRange = sheet.getRange(beginningRow, COLUMN_NUMBER.APPROVAL,
+      lastRow - frozenRows, 1);
+  var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues)
+      .build();
+  approvalColumnRange.setDataValidation(rule);
+  approvalColumnRange.setValue('IN PROGRESS');
+
+  // adds new notified column
+  sheet.insertColumnAfter(COLUMN_NUMBER.APPROVAL); // global
+  sheet.getRange(1, COLUMN_NUMBER.APPROVAL + 1).setValue('NOTIFIED STATUS');
+
+  // make sure notified column is all drop-down menus
+  var notifiedColumnRange = sheet.getRange(beginningRow, COLUMN_NUMBER.APPROVAL
+      + 1, lastRow - frozenRows, 1);
+  dropdownValues = ['NOTIFIED', 'NOT NOTIFIED'];
+  rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues)
+      .build();
+  notifiedColumnRange.setDataValidation(rule);
+  notifiedColumnRange.setValue('NOT NOTIFIED');
+}
+
+/** Runs the script "Send Emails". Sends e mails to every employee notifying
+ * them whether or not their timesheet has been approved.  Checks + updates
+ * "NOTIFIED" status so employers do not send e mails more than once.
+*/
+function sendEmails() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  // lastCol here is the NOTIFIED column
+  var frozenRows = sheet.getFrozenRows();
+  var beginningRow = frozenRows + 1;
+
+  // gets ranges of values all at once
+  var emailValues = sheet.getRange(beginningRow, COLUMN_NUMBER.EMAIL, lastRow
```

Comment:
As mentioned earlier, calculate `lastRow - frozenRows` once, and use it on all of these ranges.